core/tasks/realtime/ctasks/event_received.go

```diff
@@ -128,6 +128,20 @@ func (t *EventReceivedTask) handle(ctx context.Context, rt *runtime.Runtime, oa
 		}
 
 		if trig != nil {
+			if t.EventType == models.EventTypeNewConversation || t.EventType == models.EventTypeReferral {
```

Comment:
I don't think this logic is contingent on there being a matching trigger... and also putting it here means the engine contact isn't updated.

I would move it up to just before where we create the engine contact like for `msg_received` tasks.